src/main/java/space/obminyashka/items_exchange/service/impl/SendGridService.java

```diff
@@ -76,15 +76,20 @@ public UUID sendEmailTemplateAndGenerateConfrimationCode(String emailTo, EmailTy
         return codeId;
     }
 
+
     private static Personalization createPersonalizationAndSetParameters(EmailType emailType, UUID codeId, String host) {
         var personalization = new Personalization();
 
-        EMAIL_TEMPLATE_KEYS.forEach((key, value)->{
+        EMAIL_TEMPLATE_KEYS.forEach((key, value) -> {
             var parameterSource = emailType.name().toLowerCase().concat(".").concat(value);
             personalization.addDynamicTemplateData(key, getMessageSource(parameterSource));
         });
 
-        personalization.addDynamicTemplateData("url", host.concat(EMAIL_VALIDATE_CODE.replace("{code}", codeId.toString())));
+        if (emailType.equals(EmailType.CHANGING)) {
+            personalization.addDynamicTemplateData("url", host.concat(EMAIL_VALIDATE_CODE.replace("{code}", codeId.toString())));
+        } else if (emailType.equals(EmailType.RESET)) {
+            personalization.addDynamicTemplateData("url", host.concat(USER_SERVICE_PASSWORD_CONFIRM));
+        }
```

Comment:
It can be replaced by new map for url without using if

Reply:
if we add new endpoint for working with emails, we will need to add new clause to if and this will violate SOLID

Reply:
Changed